apps/gp2-server/src/routes/user.route.ts

```diff
@@ -104,6 +106,37 @@ export const userRouteFactory = (userController: UserController): Router =>
 
         res.json(userResponse);
       },
-    );
+    )
+    .post('/users/:userId/avatar', async (req, res) => {
+      const { params, body, loggedInUser } = req;
+      const { userId } = validateUserParameters(params);
+
+      // user is trying to update someone else
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      if (loggedInUser!.id !== userId) {
+        throw Boom.forbidden();
+      }
+      const payload = validateUserPostRequestInput(body);
+      const parsed = parseURI(payload.avatar);
+      if (!parsed) {
+        throw Boom.badRequest('avatar must be a valid data URL');
+      }
+      if (!parsed.contentType.startsWith('image')) {
+        throw Boom.unsupportedMediaType('Content-type must be image');
+      }
+      const avatar = parsed.toBuffer();
+      // convert bytes to MB and check size
+      // 3MB = 2.8MB (2MB Base64 image) + some slack
+      if (avatar.length / 1e6 > 3) {
+        throw Boom.entityTooLarge('Avatar must be smaller than 2MB');
+      }
+      const result = await userController.updateAvatar(
+        userId,
+        avatar,
+        parsed.contentType,
+        userId,
```

Comment:
Shouldn't be loggedInUser.id instead of userId? Seems a lil bit odd to use it twice. Maybe change it here or refactor the function in the userController?